rwengine/src/ai/CharacterController.cpp

```diff
@@ -252,12 +252,12 @@ bool Activities::EnterVehicle::update(CharacterObject *character,
                     character->getCurrentState().armour = 100.f;
                 } else if (vehicle->getVehicle()->vehiclename_ == "AMBULAN") {
                     // Receive up to 25 HPs from Ambulance
-                    if (character->getCurrentState().health < 100.f) {
-                        if (character->getCurrentState().health > 75.f) {
-                            character->getCurrentState().health = 100.f;
-                        } else {
-                            character->getCurrentState().health += 25.f;
-                        }
+                    float health = character->getCurrentState().health;
+                    // todo: Replace by correct fullHealth once implemented
+                    float fullHealth = 100.f;
+                    if (health < fullHealth) {
+                        character->getCurrentState().health =
+                            (health >= 80.f) ? fullHealth : health + 20.f;
```

Comment:
I've tested a few times and the game mechanics of GTA-III never raises the health above 100 from ambulances, it does waste the bonus inside that car though, so the second time the player would enter the vehicle the bonus would not get triggered even if the player's health will be below 100.